### PR TITLE
Update external db time

### LINKED ISSUE #91: Update User Timer Every 30 Seconds
Timer - Update logged in user's timer. Update timer every 30s

### PR DESCRIPTION
**Pull request by JesseyH**
Updates external db time by set interval at every interval (currently set to 15 seconds)

resolves #91

### CODE REVIEW

File: project/ConsoleTracker/ConsoleTrackerApp/tasks.py [hunk 1]
```diff
@@ -82,3 +130,7 @@ def start_query_daemon():
     t = Thread(target=query_time)
     t.daemon = True
     t.start()
+
+    t2 = Thread(target=update_time_thread)
+    t2.daemon = True
+    t2.start()
```
Comment (Thanh2311) [inline project/ConsoleTracker/ConsoleTrackerApp/tasks.py:136]: is it better to create another daemon for this? 
someone added another periodic function within the existing background thread, so shouldn't we do the same here

Comment (JesseyH) [reply project/ConsoleTracker/ConsoleTrackerApp/tasks.py:136]: I didn't want to add to the complexity of the existing function (also didn't want to touch your code, lol). I didn't want to make the code too difficult to follow, but there is reason to keep it in the same thread. With how I currently have it set up I'm worried about race conditions on accessing the expired data on the User_uses_machine model.

I can go ahead and refactor the code to separate the different tasks in the thread so it looks cleaner. Let me know what you think.

File: project/ConsoleTracker/ConsoleTrackerApp/tasks.py [hunk 2]
```diff
@@ -27,6 +31,50 @@ def switch_on(ip):
...
+def update_time_thread():
+    #represents the seconds on the clock we will update time
+    check_seconds = [00, 15, 30, 45]
+    interval = 15
+    #create the socket that connects to ExternalSocket
+    socket = InternalSocket()
+
+    while True:
+        #The current system time and the seconds of the current clock
+        curr_time = datetime.datetime.now().replace(tzinfo=pytz.UTC)
```
Comment (Thanh2311) [inline project/ConsoleTracker/ConsoleTrackerApp/tasks.py:48]:
```suggestion
        curr_time = timezone.now()
```
getting weird numbers with the original
image
using the same django's timezone class used for machine.start_time gives normal looking numbers
image

File: project/ConsoleTracker/ConsoleTrackerApp/tasks.py (hunk 1 above)
Comment (Thanh2311) [reply project/ConsoleTracker/ConsoleTrackerApp/tasks.py:136]: Yea that makes sense. Refactoring the code that looks for expired data into a different function, then adding your update time function before it sounds good here.

**Review by Thanh2311 — APPROVED** [review]
There are some issues, but i'll fix them in new PR

**Pull request closed without merging**